### PR TITLE
Logical flow fixes

### PR DESCRIPTION
**Pull request by nev-r**
fixes some of the warnings here https://lgtm.com/projects/g/DestinyItemManager/DIM/

### CODE REVIEW

File: src/app/inventory/reducer.ts [hunk 1]
```diff
@@ -227,22 +227,6 @@ function computeNewItems(oldStores: DimStore[], oldNewItems: Set<string>, newSto
     }
   }
 
-  let equal = true;
```
Comment (bhollis) [inline src/app/inventory/reducer.ts:230]: wow, can't believe I left this here

File: src/app/loadout/LoadoutPopup.tsx [hunk 2]
```diff
@@ -308,14 +308,12 @@ function LoadoutPopup({
                 </span>
               )}
             </li>
-            {!hideFarming && (
```
Comment (bhollis) [inline src/app/loadout/LoadoutPopup.tsx:311]: I don't follow this one - what is hideFarming even for?

**Review by bhollis — APPROVED** [review]

File: src/app/loadout/LoadoutPopup.tsx (hunk 2 above)
Comment (delphiactual) [reply src/app/loadout/LoadoutPopup.tsx:311]: It was added with Active Mode 
https://github.com/DestinyItemManager/DIM/pull/5969 @kyleshay

Comment (nev-r) [reply src/app/loadout/LoadoutPopup.tsx:311]: it's bool checked upstream already is all i know

**Pull request merged**